dali/base/dafdesc.cpp

```diff
@@ -1702,6 +1733,30 @@ class CFileDescriptor:  public CFileDescriptorBase, implements ISuperFileDescrip
         return *attr.get();
     }
 
+    virtual IPropertyTree *getHistory()
```

Comment:
To avoid code duplication, you may as well impl. getHistory by calling queryHistory() .